src/pybind/mgr/cephadm/serve.py

```diff
@@ -618,9 +621,31 @@ def _apply_service_config(self, spec: ServiceSpec) -> None:
                             'who': section,
                         })
                     except MonCommandFailed as e:
-                        self.log.warning(
-                            f'Failed to set {spec.service_name()} option {k}: {e}'
-                        )
+                        msg = f'Failed to set {spec.service_name()} option {k}: {e}'
+                        self.log.warning(msg)
+                        warning_name = f'{spec.service_name()}_{k}_FAILED_TO_SET_OPTION'
+                        self.set_health_warning(warning_name, msg, 1, [f'{str(e)}'])
+                        options_failed_to_set.append(warning_name)
+
+            for warning in self.mgr.health_checks.copy():
+                if warning not in invalid_config_options and warning.endswith('_INVALID_CONFIG_OPTION'):
+                    self.remove_health_warning(warning)
+                elif warning not in options_failed_to_set and warning.endswith('_FAILED_TO_SET_OPTION'):
+                    self.remove_health_warning(warning)
+
+    def set_health_warning(self, name: str, summary: str, count: int, detail: List[str]) -> None:
+        self.mgr.health_checks[name] = {
+            'severity': 'warning',
+            'summary': summary,
+            'count': count,
+            'detail': detail,
+        }
+        self.mgr.set_health_checks(self.mgr.health_checks)
+
+    def remove_health_warning(self, name: str) -> None:
+        if name in self.mgr.health_checks:
+            del self.mgr.health_checks[name]
+            self.mgr.set_health_checks(self.mgr.health_checks)
```

Comment:
I like these helpers.  Can you move them to module.py and put them in a separate patch, along with the changes to existing callers?

Reply:
To build on this @melissa-kun-li , these are the checks I could see that we set that could make use of your helper if you're interested in setting that up.

CEPHADM_PAUSED (you already got this one)

CEPHADM_STRAY_DAEMON / CEPHADM_STRAY_HOST

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L515-L516
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L562-L579

CEPHADM_HOST_CHECK_FAILED / CEPHADM_FAILED_DAEMON /  CEPHADM_REFRESH_FAILED

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L325-L364

HOST_IN_MAINTENANCE

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/module.py#L1581-L1591

There's also a bunch of them set in configchecks.py and upgrade.py but those are being set/removed a bit differently to work with the process for config checks and upgrade respectively so should probably be left as is.

Reply:
@liewegas @adk3798 sure, on it !